app/inbox/views.py

```diff
@@ -0,0 +1,148 @@
+import json
+from inbox.models import Notification
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth import get_user_model
+from django.http import JsonResponse, HttpResponseForbidden
+from django.core.paginator import Paginator
+from django.views.decorators.csrf import csrf_exempt
+
+
+def notifications(request):
+    """Handle all notifications."""
+    if request.method == 'GET':
+        profile = request.user.profile if request.user.is_authenticated and request.user.profile else None
+        if profile is None:
+            return HttpResponseForbidden('Not Allowed')
+        limit = 10
+        page = 1
+        if 'limit' in request.GET:
+            limit = int(request.GET['limit'])
+        if 'page' in request.GET:
+            page = int(request.GET['page'])
+        print(limit, page)
+        all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('id')[::-1]
+        params = dict()
+        all_pages = Paginator(all_notifs, limit)
+        if page > 0 and page <= all_pages.num_pages:
+            all_notifications = []
+            for i in all_pages.page(page):
+                new_notif = i.to_standard_dict()
+                new_notif['created_on'] = i.created_on
+                new_notif['modified_on'] = i.modified_on
+                new_notif['username'] = get_user_model().objects.get(id=new_notif['from_user_id']).username
```

Comment:
No need to do this - You already have the user relation on `Notification` - so:

`username = i.from_user.username` following updates above

Realistically, we should move this into the dictionary handling since we already have `User` in both cases.